server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -149,6 +153,22 @@ private static int effectiveRoutingToHash(String effectiveRouting) {
      */
     public void checkIndexSplitAllowed() {}
 
+    /**
+     * If this index is in the process of resharding, and the shard to which this request is being routed,
+     * is a target shard that is not yet in HANDOFF state, then route it to the source shard.
+     * @param shardId  shardId to which the current document is routed based on hashing
+     * @return Updated shardId
+     */
+    protected final int rerouteIfResharding(int shardId) {
+        if (indexReshardingMetadata != null && indexReshardingMetadata.getSplit().isTargetShard(shardId)) {
```

Comment:
nit: should we check for `indexReshardingMetadata.isSplit()`? It's technically possible.

Reply:
➕

Reply:
possibly we should assert it is split, because I don't know right now what the correct behaviour here would be if it isn't. Good chance that we'll also need some logic here for any other resharding type.

Reply:
yes i am okay with assert